front/src/modules/simulationResult/types.ts

```diff
@@ -18,6 +17,10 @@ import type { PacedTrainWithDetails } from 'modules/trainschedule/components/Tim
 import type { OccurrenceId, TimetableItem, TrainScheduleId } from 'reducers/osrdconf/types';
 import type { ArrayElement } from 'utils/types';
 
+export type PathOperationalPoint = NonNullable<PathProperties['operational_points']>[number] & {
+  opId: string | null;
+};
```

Comment:
I think defining a new type is a great idea! However, I find it confusing to silently re-purpose the `id` field.

In the original `PathProperties['operational_points']` type, the `id` field is an OP ID. In `PathOperationalPoint`, the `id` field is a waypoint ID.

I would suggest to either:

- Leave the `id` field from the original type with the same semantics (OP ID), drop `opId`, and add a new separate field for the waypoint ID.
- Omit the `id` field from `PathOperationalPoint`, keep `opId`, and define a new, different separate field for the waypoint ID.

If these options are too much work, we could also add a doc comment explaining this subtlety, but I'd still find this surprising and inconsistent with editoast.

Reply:
I agree, I went that direction because of the impact. I'll check again how much work it would require to store the custom waypoint ID in another field.